libs/server/Objects/List/ListObjectImpl.cs

```diff
@@ -187,9 +187,15 @@ private void ListRange(byte* input, ref SpanByteAndMemory output)
                         while (!RespWriteUtils.WriteArrayLength(count, ref curr, end))
                             ObjectUtils.ReallocateOutput(ref output, ref isMemory, ref ptr, ref ptrHandle, ref curr, ref end);
 
-                        for (int i = start; i <= stop; i++)
+                        var i = -1;
+                        foreach (var bytes in list)
                         {
-                            while (!RespWriteUtils.WriteBulkString(list.ElementAt(i), ref curr, end))
+                            i++;
```

Comment:
These offsets can also be negative numbers indicating offsets starting at the end of the list. They can also be larger than the number of elements in the list.

For example, -1 is the last element of the list, -2 the penultimate, and so on. See https://redis.io/commands/lrange/.

So if a start or end offset is -X, you should convert it to `Max(list.Count - X, 0)`
If a start of end offset is non-negative X, you should convert it to `Min(X, list.Count - 1)`

Test case would be: `LRANGE mylist -100 100`

Reply:
If start is larger than the end of the list, an empty list is returned. If stop is larger than the actual end of the list, Redis will treat it like the last element of the list.

Reply:
> These offsets can also be negative numbers indicating offsets starting at the end of the list. They can also be larger than the number of elements in the list.
> 
> For example, -1 is the last element of the list, -2 the penultimate, and so on. See https://redis.io/commands/lrange/.
> 
> So if a start or end offset is -X, you should convert it to `Max(list.Count - X, 0)` If a start of end offset is non-negative X, you should convert it to `Min(X, list.Count - 1)`
> 
> Test case would be: `LRANGE mylist -100 100`



`start` and `stop` have been fixed to a positive number in previous code, and test is ok

```
127.0.0.1:3278> lpush list 1 2 3 4 5
(integer) 5
127.0.0.1:3278> lrange list 0 99
1) "5"
2) "4"
3) "3"
4) "2"
5) "1"
127.0.0.1:3278> lrange list -100 100
1) "5"
2) "4"
3) "3"
4) "2"
5) "1"
127.0.0.1:3278> lrange list -3 -1
1) "3"
2) "2"
3) "1"
127.0.0.1:3278> lrange list 2 4
1) "3"
2) "2"
3) "1"
```